lib/internal/worker.js

```diff
@@ -506,6 +506,43 @@ class Worker extends EventEmitter {
       };
     });
   }
+
+  // TODO(theanarkh): add options, such as sample_interval, CpuProfilingMode
+  startCpuProfile(name) {
+    validateString(name, 'name');
+    const startTaker = this[kHandle]?.startCpuProfile(name);
+    return new Promise((resolve, reject) => {
+      if (!startTaker) return reject(new ERR_WORKER_NOT_RUNNING());
+      startTaker.ondone = (err) => {
+        if (err) {
+          return reject(err);
+        }
+        let stopped = false;
+        const stop = () => {
+          if (stopped) {
+            return;
+          }
+          stopped = true;
+          const stopTaker = this[kHandle]?.stopCpuProfile(name);
+          return new Promise((resolve, reject) => {
```

Comment:
```suggestion
            return stopped;
          }
          const stopTaker = this[kHandle]?.stopCpuProfile(name);
          return stopped = new Promise((resolve, reject) => {
```

This would ensure that if there are multiple calls to `stop()`, they do not only not interfere with each other, they actually return the same promise and therefore all resolve at the same time with the same value

Reply:
Done !